src/lava/magma/runtime/runtime_service.py

```diff
@@ -153,6 +206,30 @@ def _relay_pm_ack_given_model_id(self, model_id: int):
         ack_relay_port = self.service_to_runtime
         ack_relay_port.send(ack_recv_port.recv())
 
+    def _handle_pause(self):
```

Comment:
PAUSE/HOST can only be entered at the end of a time step. We need to put the system in a safe state because we might want to resume.